### DISCUSSION TITLE
Disabled/required fields are always validated by zod

### DISCUSSION THREAD

**Discussion by viveleroi** [react-hook-form/react-hook-form#9378] [General]
I'm using RHF and the zod resolver. Zod assumes that all schema values are required by default. I was hoping the rhf resolver could somehow override that when a field is not required, or is disabled.

I've tried applying both attributes to the `input` itself as well as the `register` function, but neither work.

Is there a way I can get rhf/zod to ignore the validation if a field is not required and empty, or it's disabled?

```jsx
      <input
        {...register(name, {
          ...registerOpts,
          disabled,
          required
        })}
        className={clsx(styles.input, { [styles.error]: (isTouched || isSubmitted) && error })}
        disabled={disabled}
        required={required}
      />
```

**Comment by bluebill1049** [comment]
This is possible with build validation, however with disabled input schema will still receive the entire form values, you may want to keep an extra state with disabled input and send the `context` object into your schema to conditionally validate those disabled inputs.
